### ISSUE TITLE
Use Argon2 for all key derivation functionality

### ISSUE THREAD

**Issue by ricardobalk** [ricardobalk/vault#12]
Argon2 has been added for a better way of key derivation and verification. Better means: Instead of relying on the amount of iterations, which PBKDF2 does, you can also tell Argon2 how much memory and how many threads to use, making it a more modern (and secure) key derivation method. In addition to this, Argon2 also allows extra data to be used for its input, allowing users to provide key files in addition to a passphrase.

TL;DR version - Anyway, at this particular moment, Argon2 is only used as a way for generating and verifying password hashes. The goal is to replace PBKDF2 by Argon2, and use Argon2 for both the verification and deriving encryption keys.
